subsystems/sonarSensor.py

```diff
@@ -11,10 +11,11 @@ class SonarSensor:
     Example:
         sonarSensor = SonarSensor('10.10.76.11', 5811)
     """
-    def __init__(self, sensor_ip, listen_port, logger=None):
+    def __init__(self, sensor_ip, listen_port, simulation=False, logger=None):
```

Comment:
One more suggestion.    I think because you've put the sim/nosim switch inside, you can do the check in the constructor instead of in the createChannel function.  Just check, save the appropriate values in the instance variables, and then just use the self.ip and self.port in the createChannel().